samcli/commands/delete/delete_context.py

```diff
@@ -0,0 +1,141 @@
+        self.s3_bucket = None
+        self.s3_prefix = None
+        self.cf_utils = None
+        self.s3_uploader = None
+        self.cf_template_file_name = None
+        self.delete_artifacts_folder = None
+        self.delete_cf_template_file = None
+
+    def __enter__(self):
+        self.parse_config_file()
+        if not self.stack_name:
+            self.stack_name = prompt(
+                click.style("\tEnter stack name you want to delete:", bold=True), type=click.STRING
+            )
+
+        if not self.region:
+            self.region = prompt(click.style("\tEnter region you want to delete from:", bold=True), type=click.STRING)
+        return self
+
+    def __exit__(self, *args):
+        pass
+
+    def parse_config_file(self):
+        """
+        Read the provided config file if it exists and assign the options values.
+        """
+        toml_provider = TomlProvider(CONFIG_SECTION, [CONFIG_COMMAND])
+        config_options = toml_provider(
+            config_path=self.config_file, config_env=self.config_env, cmd_names=[CONFIG_COMMAND]
+        )
+        if config_options:
+            if not self.stack_name:
+                self.stack_name = config_options.get("stack_name", None)
+            if self.stack_name == config_options["stack_name"]:
+                if not self.region:
+                    self.region = config_options.get("region", None)
+                if not self.profile:
+                    self.profile = config_options.get("profile", None)
+                self.s3_bucket = config_options.get("s3_bucket", None)
+                self.s3_prefix = config_options.get("s3_prefix", None)
+
+    def delete(self):
+        """
+        Delete method calls for Cloudformation stacks and S3 and ECR artifacts
+        """
+        template_str = self.cf_utils.get_stack_template(self.stack_name, TEMPLATE_STAGE)
+
+        if self.s3_bucket and self.s3_prefix:
+            self.delete_artifacts_folder = confirm(
+                click.style(
+                    "\tAre you sure you want to delete the folder"
+                    + f" {self.s3_prefix} in S3 which contains the artifacts?",
+                    bold=True,
+                ),
+                default=False,
+            )
+            if not self.delete_artifacts_folder:
+                with mktempfile() as temp_file:
+                    self.cf_template_file_name = get_cf_template_name(temp_file, template_str, "template")
+                self.delete_cf_template_file = confirm(
+                    click.style(
+                        "\tDo you want to delete the template file" + f" {self.cf_template_file_name} in S3?", bold=True
+                    ),
+                    default=False,
+                )
+
+        click.echo("\n")
+        # Delete the primary stack
+        self.cf_utils.delete_stack(stack_name=self.stack_name)
+
+        click.echo(f"- deleting Cloudformation stack {self.stack_name}")
```

Comment:
Is this consistent with the UI conventions in SAM CLI? (colors, indentation, capitalization) I know it's a bit of a mess currently, but until we have SAM CLI-wide UI guidelines, we should strive for consistency so the experience feels coherent.